core/src/Streamly/Internal/Data/Serialize/TH.hs

```diff
@@ -232,10 +242,40 @@ mkDeserializeExpr False headTy tyOfTy =
                         $(lift (pprint headTy)) ++ ")")|])
             []
 
+mkDeserializeExpr False True headTy (TheType con@(SimpleDataCon _ fields)) = do
+    deserializeWithKeys <- newName "deserializeWithKeys"
+    updateFunc <- newName "updateFunc"
+    hOff <- newName "hOff"
+    finalOff <- newName "finalOff"
+    deserializeWithKeysMethod <-
+        RecHeader.mkDeserializeKeysExpr
+            hOff finalOff _arr _endOffset updateFunc con
+    updateFuncDec <- RecHeader.conUpdateFuncExpr updateFunc headTy fields
+    letE
```

Comment:
This is the slow path. Move these out into a where clause with a `NOINLINE` pragma.